plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ScaleDeploymentHandler.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.handler;
+
+import java.util.Collection;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.math.NumberUtils;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.eclipse.jface.dialogs.InputDialog;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.core.connection.ConnectionsRegistryUtil;
+import org.jboss.tools.openshift.internal.common.ui.utils.UIUtils;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.models.Deployment;
+import org.jboss.tools.openshift.internal.ui.models.IResourceUIModel;
+
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IReplicationController;
+
+/**
+ * Handle for scaling deployments
+ * @author jeff.cantrill
+ *
+ */
+public class ScaleDeploymentHandler extends AbstractHandler{
+
+	public static final String REPLIA_DIFF = "org.jboss.tools.openshift.ui.command.deployment.scale.replicadiff";
+	
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		IReplicationController rc = getSelectedElement(event, IReplicationController.class);
+		if(rc == null) {
+			rc = getSelectedElement(event, IDeploymentConfig.class);
+		}
+		if(rc != null) {
+			scaleUsing(event, rc, rc.getName());
+			return null;
+		}
+		Deployment deployment = getSelectedElement(event, Deployment.class);
```

Comment:
Didn't you say you would allow the deployment to scale independantly from the original deployment config replica#, ie. if deployment set to scale to 1, from a deployment config set to 2, then redeploying the service would respin 2 replicas